SkyDrive/SkyDrive/Controllers/InstructorController.cs

```diff
@@ -30,5 +44,35 @@ public InstructorEntity Post(InstructorEntity instructorEntity)
 
             return instructorEntity;
         }
+
+        [HttpPut]
+        public InstructorEntity Put(InstructorEntity instructorEntity)
+        {
+            var entity = _context.Instructors.Find(instructorEntity.Id);
+
+            if (entity == null)
+            {
+                throw new EntityNotFoundException($"Instructor with id: {instructorEntity.Id} not found");
+            }
+
+            _context.Instructors.Update(instructorEntity);
+            _context.SaveChanges();
+
+            return instructorEntity;
+        }
+
+        [HttpDelete("{id:int}")]
+        public void Delete(int id)
+        {
+            var entity = _context.Instructors.Find(id);
+
+            if (entity == null)
+            {
+                throw new EntityNotFoundException($"Instructor with id: {id} not found");
+            }
+
```

Comment:
Possibly makes sense to move an entity exists checking logic to a separate auxiliary method, smth like IsEntityExists(int id), later you can move that method to your future services or repositories.